src/Evolution/Systems/GeneralizedHarmonic/UpwindPenaltyCorrection.hpp

```diff
@@ -0,0 +1,293 @@
+ *   \lambda_{v^0} =& -\beta^i n_i -v^i_g n_i, \\
+ *   \lambda_{v^\pm} =& \pm \alpha - \beta^i n_i - v^i_g n_i,
+ * \f}
+ *
+ * where \f$v_g^i\f$ is the mesh velocity. We have also defined
+ *
+ * \f{align}{
+ *   \tilde{\lambda}_{\hat{\alpha}} =
+ *    \left\{
+ *        \begin{array}{ll}
+ *          \lambda_{\hat{\alpha}} &
+ *            \mathrm{if}\;\lambda_{\hat{\alpha}}\le 0.0 \\
+ *          0 & \mathrm{otherwise}
+ *        \end{array}\right.
+ * \f}
+ *
+ * Note that we have assumed \f$n_i^{\mathrm{ext}}\f$ points in the same
+ * direction as \f$n_i^{\mathrm{int}}\f$, but in the code we have them point in
+ * opposite directions. If \f$n_i^{\mathrm{ext}}\f$ points in the opposite
+ * direction the external speeds have their sign flipped and the \f$\pm\f$
+ * fields and speeds reverse roles. Specifically, in the code we have:
+ *
+ * \f{align}{
+ *   \label{eq:pnpm upwind gh metric code}
+ *   D_{g_{ab}} &= \bar{\lambda}_{v^{g}}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},g}_{ab}
+ *                - \tilde{\lambda}_{v^{g}}^{\mathrm{int}}
+ *                v^{\mathrm{int},g}_{ab}, \\
+ *   \label{eq:pnpm upwind gh pi code}
+ *   D_{\Pi_{ab}}
+ *              &= \frac{1}{2}\left(\bar{\lambda}_{v^+}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},+}_{ab} +
+ *                \bar{\lambda}_{v^-}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},-}_{ab}\right)
+ *                + \bar{\lambda}_{v^g}^\mathrm{ext}\gamma_2
+ *                v^{\mathrm{ext},g}_{ab}
+ *                \notag \\
+ *              &-\frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{int}}
+ *                v^{\mathrm{int},+}_{ab} +
+ *                \tilde{\lambda}_{v^-}^{\mathrm{int}}
+ *                v^{\mathrm{int},-}_{ab}\right)
+ *                - \tilde{\lambda}_{v^g}^\mathrm{int}\gamma_2
+ *                v^{\mathrm{int},g}_{ab} , \\
+ *   \label{eq:pnpm upwind gh phi code}
+ *   D_{\Phi_{iab}}
+ *              &= \frac{1}{2}\left(\bar{\lambda}_{v^+}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},+}_{ab}
+ *                - \bar{\lambda}_{v^-}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},-}_{ab}\right)n_i^{\mathrm{ext}}
+ *                + \bar{\lambda}_{v^0}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},0}_{iab}
+ *                \notag \\
+ *              &-
+ *                \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{int}}
+ *                v^{\mathrm{int},+}_{ab}
+ *                - \tilde{\lambda}_{v^-}^{\mathrm{int}}
+ *                v^{\mathrm{int},-}_{ab}\right)n_i^{\mathrm{int}}
+ *                - \tilde{\lambda}_{v^0}^{\mathrm{int}}
+ *                v^{\mathrm{int},0}_{iab},
+ * \f}
+ *
+ * where
+ *
+ * \f{align}{
+ *   \bar{\lambda}_{\hat{\alpha}} =
+ *    \left\{
+ *        \begin{array}{ll}
+ *          -\lambda_{\hat{\alpha}} &
+ *            \mathrm{if}\;-\lambda_{\hat{\alpha}}\le 0.0 \\
+ *          0 & \mathrm{otherwise}
+ *        \end{array}\right.
+ * \f}
+ */
```

Comment:
Thanks! I presume the equations are correct, because the code works empirically, but I'm confused by the "+/- swap roles)...I would have expected this would give the opposite sign in the () terms for D_{\Phi} in the spectre version vs the "same direction normals" version?

Reply:
This comes up because of the extra minus sign in the `\bar{\lambda}`, which is how it's written in the code and why I wrote it this way in the documentation